feat(plugin-chart-echarts): Echarts Waterfall

stephenLYZ opened the pull request:
🏆 Enhancements

Waterfall charts are useful for understanding how an initial value (like net income) is affected by a series of positive and negative changes. 

For example, we can see the trend of personal annual income. Here we use the year as the `series`.
![image](https://user-images.githubusercontent.com/11830681/120924226-1d94ff80-c705-11eb-8adf-a7da594a5599.png)

Moreover, we can use the value in `breakdowns` to add additional data to the chart. It adds the contributors to increases or decreases for each series.
![image](https://user-images.githubusercontent.com/11830681/120893346-68e7d900-c645-11eb-8afe-e55290cbf754.png)

This pr refers to #705 (Thanks for Simcha's work.

stephenLYZ committed:
fix(plugin-chart-echarts): waterfall

stephenLYZ committed:
fix(plugin-chart-echarts): rename

stephenLYZ committed:
fix(plugin-chart-echarts): remove some unused code

stephenLYZ committed:
fix(plugin-chart-echarts): remove some unused code

stephenLYZ committed:
fix(plugin-chart-echarts): add unit test

stephenLYZ committed:
fix(plugin-chart-echarts): change transformProps 逻辑

stephenLYZ committed:
fix(plugin-chart-echarts): ci

xiezhongfu commented:
Thanks stephen.
Core-code is LGTM for me. Next, I will review the code in detail.

zhaoyongjie reviewed (commented):
Awesome works! Leave some issues.

1. the `null` value needs to be handled in tooltip
![image](https://user-images.githubusercontent.com/2016594/126864477-292e32a3-0898-47fa-8fd4-bae1747dd242.png)


2. please use `seriese` control as category, and need to DND support
![image](https://user-images.githubusercontent.com/2016594/126864528-f001ab95-5e8e-492d-9743-822df56c8cfe.png)

3. I can't hidden `metric value` and `tooltip` on the viz

![image](https://user-images.githubusercontent.com/2016594/126864683-25c17e19-ef73-4d4e-9833-d40754ef1011.png)

4. the `Breakdown` is broken
<img width="1514" alt="image" src="https://user-images.githubusercontent.com/2016594/126864857-26911910-df1a-48ea-bcdb-2be046836bb5.png">

zhaoyongjie commented on plugins/plugin-chart-echarts/src/Waterfall/index.ts:
```diff
@@ -0,0 +1,55 @@
+      buildQuery,
+      controlPanel,
+      loadChart: () => import('./EchartsWaterfall'),
+      metadata: new ChartMetadata({
```
needs category property here. Might `Ranking`  is  is an appropriate category

rusackas replied:
`Evolution` seems reasonable to me.

stephenLYZ committed:
feat: fix some problems

stephenLYZ committed:
fix: test

stephenLYZ commented:
> Awesome works! Leave some issues.
> 
> 1. the `null` value needs to be handled in tooltip
>    ![image](https://user-images.githubusercontent.com/2016594/126864477-292e32a3-0898-47fa-8fd4-bae1747dd242.png)
> 2. please use `seriese` control as category, and need to DND support
>    ![image](https://user-images.githubusercontent.com/2016594/126864528-f001ab95-5e8e-492d-9743-822df56c8cfe.png)
> 3. I can't hidden `metric value` and `tooltip` on the viz
> 
> ![image](https://user-images.githubusercontent.com/2016594/126864683-25c17e19-ef73-4d4e-9833-d40754ef1011.png)
> 
> 1. the `Breakdown` is broken
> 
> <img alt="image" width="1514" src="https://user-images.githubusercontent.com/2016594/126864857-26911910-df1a-48ea-bcdb-2be046836bb5.png">

1. fixed.
2. fixed.
3. add `show_value` control and `rich_tooltip` control for this chart. I'm not sure 
4. The logic of the previous breakdown was to hide the breakdown of the first series and only show the total (a reference to powerBI?). Maybe it is strange because it will cause the data and the chart to not correspond, I removed this special logic first and need to discuss whether this special logic is needed.

stephenLYZ committed:
fix: lint

villebro commented:
The codebase on this repo has been moved to the main [Apache Superset repo](https://github.com/apache/superset), and consequently the repo is in the process of being archived. See the Superset Improvement Proposal for details: https://github.com/apache/superset/issues/13013 . While all currently open issues and PRs will be closed, we encourage you to reopen this PR on the main repo, which should be as simple as moving over any code changes as follows:
- core packages: `packages/**` (this repo) -> `superset-frontend/packages/**` (main repo)
- plugins: `plugins/**` (this repo) -> `superset-frontend/plugins/**` (main repo)

If you need help with the migration, please post a message on the SIP or reach out on the [community Slack](https://join.slack.com/t/apache-superset/shared_invite/zt-uxbh5g36-AISUtHbzOXcu0BIj7kgUaw).
